zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -120,12 +129,31 @@ public PrepRequestProcessor(ZooKeeperServer zks, RequestProcessor nextProcessor)
             + "):", zks.getZooKeeperServerListener());
         this.nextProcessor = nextProcessor;
         this.zks = zks;
+        initMaxEphemeralNodesPerSession();
         this.digestEnabled = ZooKeeperServer.isDigestEnabled();
         if (this.digestEnabled) {
             this.digestCalculator = new DigestCalculator();
         }
     }
 
+    private void initMaxEphemeralNodesPerSession() {
+        this.ephemeralNodesPerSessionMap = new ConcurrentHashMap<>();
+        maxEphemeralNodesPerSession = Integer.getInteger(ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION, ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION_DEFAULT);
+        if (maxEphemeralNodesPerSession < 0) {
```

Comment:
Do we need to keep the possibility of not throttling? I specifically mean that if property value of `ZOOKEEPER_MAX_EPHEMERAL_COUNT_PER_SESSION` is set to negative, should we honor it?